example/deploy-docs-live-code-contracts.ts

```diff
@@ -35,80 +35,95 @@ export const tezos = new TezosToolkit(provider);
-        console.log(`Transfer complete`);
-      } else {
-        console.log(`${key} has an ok balance of ${user_balance / 1000000}.`);
+        low_balance.push(users[i]);
       }
+    }
+  } catch (ex) {
+    console.error(ex);
+  }
 
-      async function transferMinBalance() {
-        await tezos.contract
-          .transfer({ to: value, amount: min_balance })
-          .then((op) => {
-            console.log(`Waiting for ${op.hash} of transfer to ${value} to be confirmed...`);
-            return op.confirmation(1).then(() => op.hash);
-          })
-          .catch((error) => console.log(`Error: ${error} ${JSON.stringify(error, null, 2)}`));
-      }
-    });
+  console.log(`Low balance addresses : ` + low_balance);
+  tezos.setSignerProvider(signer);
+  try {
+    for (var i = 0; i < low_balance.length; i++) {
+      console.log("Funding low balance address :"+low_balance[i])
+      const fundAccountFirst = await tezos.contract.transfer({
+        to: low_balance[i],
+        amount: min_balance / 1000000,
+      });
+      await fundAccountFirst.confirmation();
+
+    }
   } catch (ex) {
     console.error(ex);
   }
 }
 
 async function originateTheContracts() {
-  checkBalances();
-
-  console.log('originating...');
-  contract_catalogue.set('IncrementContract', await originateIncrementContract());
-  contract_catalogue.set('LambdaViewOne', await originateLambda1());
-  contract_catalogue.set('LambdaViewTwo', await originateLambda2());
-  contract_catalogue.set('BigMapPairasMap', await originateBigMapsPairasMapKeys());
-  contract_catalogue.set('BigMapValuesComplexKeys', await originateBigMapsComplexKeys());
-  contract_catalogue.set('BigMapInitialStorage', await originateBigMapsInitialStorage());
-  contract_catalogue.set('BigMapsMultipleValues', await originateMapValueMultipleBigMaps());
-  contract_catalogue.set(
-    'SmartContractComplexStorage',
-    await originateSmartContractComplexStorage()
-  );
-  contract_catalogue.set('Tzip12BigMapOffChain', await originateTZip12BigMapOffChain());
-  contract_catalogue.set('Tzip16Storage', await originateTzip16Storage());
-  contract_catalogue.set('Tzip16HTTPS', await originateTzip16Https());
-  contract_catalogue.set('Tzip16SHA256', await originateTzip16SHA256());
-  contract_catalogue.set('Tzip16IPFS', await originateTzip16IPFS());
-  contract_catalogue.set('Tzip16OffChainOne', await originateTzip16OnChainOne());
-  contract_catalogue.set('Tzip16OffChainTwo', await originateTzip16OnChainTwo());
-  contract_catalogue.set('WalletContract', await originateWalletOriginateContractTransfer());
-  contract_catalogue.set('WalletAreYouThereContract', await originateWalletOriginateAreYouThere());
-  contract_catalogue.set('TokenContract', await originateTokenContract());
-  contract_catalogue.set('BigMapPackContract', await originateBigMapPackContract());
-  contract_catalogue.set('MichelsonMap', await originateMichelsonMap());
-
-  json_contract_catalogue();
-
+  checkBalances(users);
```

Comment:
I suggest awaiting here or the script will start deploying contracts before the checkBalances is done.
`await checkBalances(users);`

Reply:
I have made the change, thanks